client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/AbstractPutAllPerfTest.java

```diff
@@ -0,0 +1,96 @@
+package org.infinispan.client.hotrod.stress;
+
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killRemoteCacheManager;
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killServers;
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests putAll performance with large and small data sets
+ *
+ * @author William Burns
+ * @since 7.2
+ */
+@Test
+public abstract class AbstractPutAllPerfTest extends MultipleCacheManagersTest {
+
+   protected HotRodServer[] hotrodServers;
+   protected RemoteCacheManager remoteCacheManager;
+   protected RemoteCache<Object, Object> remoteCache;
+
+   abstract protected int numberOfHotRodServers();
+
+   abstract protected ConfigurationBuilder clusterConfig();
+
+   protected final long millisecondsToRun = TimeUnit.MINUTES.toMillis(1);
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      final int numServers = numberOfHotRodServers();
+      hotrodServers = new HotRodServer[numServers];
+
+      createCluster(hotRodCacheConfiguration(clusterConfig()), numberOfHotRodServers());
+
+      for (int i = 0; i < numServers; i++) {
+         EmbeddedCacheManager cm = cacheManagers.get(i);
+         hotrodServers[i] = HotRodClientTestingUtil.startHotRodServer(cm);
+      }
+
+      String servers = HotRodClientTestingUtil.getServersString(hotrodServers);
+
+      remoteCacheManager = new RemoteCacheManager(servers);
```

Comment:
I based this test off of https://github.com/infinispan/infinispan/blob/master/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/BaseBulkGetTest.java.  I am not as familiar with exactly what I should be using here, if you want you can suggest a change to not use it.